pygmo/__init__.py

```diff
@@ -113,7 +111,143 @@ def _population_init(self, prob=None, size=0, seed=None):
 
 setattr(population, "__init__", _population_init)
 
+# Override of the island constructor.
+__original_island_init = island.__init__
+
+
+def _island_init(self, **kwargs):
+    """
+    Keyword Args:
+        udi: a user-defined island (either Python or C++ - note that *udi* will be deep-copied
+          and stored inside the :class:`~pygmo.core.island` instance)
+        algo: a user-defined algorithm (either Python or C++), or an instance of :class:`~pygmo.core.algorithm`
+        pop (:class:`~pygmo.core.population`): a population
+        prob: a user-defined problem (either Python or C++), or an instance of :class:`~pygmo.core.problem`
+        size (``int``): the number of individuals
+        seed (``int``): the random seed (if not specified, it will be randomly-generated)
+
+    Raises:
+        KeyError: if the set of keyword arguments is invalid
```

Comment:
@darioizzo I opened #53 which contains some ideas about improvements to the pygmo documentation. I think the end result in sphinx at the moment looks nice, but I noticed there are some small inconsistencies in the ipython ``?`` syntax (e.g., documentation added from C++ can sometimes look a bit different from the docs added from Python). Perhaps we can continue the discussion in #53.